webapp/data_structures/TOWS_c.js

```diff
@@ -41,33 +41,140 @@ function TOWS_c() {
-        if(this.herit == 1){
-            this.x.model.blocktype = new ScilabDouble(["x"]);
+
+        var chararray = varnam1.match(regex_char);
+        if (chararray != null) {
+            if(regex_special_withmath_char.test(varnam1)){
+                alert("Invalid variable name. \nPlease choose another variable name.");
+                throw "incorrect";
+            }
+            if(regex_parentheses.test(varnam1)){
+                alert("Invalid variable name. \nPlease choose another variable name.");
+                throw "incorrect";
+            }
+            if(regex_semicolon_comma.test(varnam1)){
+                alert("Invalid variable name. \nPlease choose another variable name.");
+                throw "incorrect";
+            }
+        }else{
+            alert("Invalid variable name. \nPlease choose another variable name.");
+            throw "incorrect";
         }
-        else{
+
+        herit1 = herit1.trim().replace(regex_parentheses, '');
+        if (!regex_num.test(herit1)) {
+            var chararray = herit1.match(regex_char);
+            if (chararray != null) {
+                if (herit1.includes("%pi")){
+                    herit1.replace("%pi", Math.PI);
+                }else{
+                    alert("Answer given for Inherit (no:0, yes:1) is incorrect:\nUndefined variable:"+herit1);
+                    throw "incorrect";
+                }
+            }
+            var specialarray = herit1.match(regex_special_char);
+            if (specialarray != null) {
+                if (herit1.includes("%pi")){
+                    herit1.replace("%pi", Math.PI);
+                }else{
+                    if(specialarray.includes("&")){
+                        alert("Answer given for Inherit (no:0, yes:1)\nhas incorrect type vec.");
+                        throw "incorrect";
+                    }else{
+                        alert("Answer given for Inherit (no:0, yes:1)\nis incorrect: Missing operator, comma, or semicolon.");
+                        throw "incorrect";
+                    }
+                }
+            }
+            var semi_comma_array = herit1.match(regex_semicolon_comma);
+            if (semi_comma_array != null) {
+                alert("Answer given for Inherit (no:0, yes:1)\nhas invalid dimension:\nwaiting for dimension 1.");
+                throw "incorrect";
+            }
+        }
+        var herit2 = parseFloat(getValueOfImaginaryInput(herit1));
+        if (herit2 != 0){
+            herit2 = 1;
+        }
+        nz2 = parseFloat(nz2);
+        this.nz = nz1;
+        this.varnam = varnam1;
+        this.herit = herit1;
+        this.x.model.intyp = new ScilabDouble([-1]);
+        var io = set_io(this.x.model,this.x.graphics,[[-1],[-2]],[],ones(1-herit2,1),[]);
+        if(herit2 == 1){
+            this.x.model.blocktype = new ScilabDouble(["x"]);
+        }else{
             this.x.model.blocktype = new ScilabDouble(["d"]);
```

Comment:
```suggestion
            this.x.model.blocktype = new ScilabString(["d"]);
```

blocktype should be string, not double